### PR TITLE
Including information about Especifcations and Providers.

### PR DESCRIPTION
**Pull request by andrelugomes**
Including information about Specifications and Providers.

Let user of this lib choose which implementation use.

Add Javax EL API.

### CODE REVIEW

File: demos/spring-5-demo/build.gradle [hunk 1]
```diff
@@ -11,14 +11,14 @@ mainClassName = 'Main'
 
 repositories {
     mavenCentral()
-    maven { url "https://oss.sonatype.org/content/repositories/releases" }
+    mavenLocal()
```
Comment (samuelgrigolato) [inline demos/spring-5-demo/build.gradle]: This forces the user to install service-validator's SNAPSHOT locally before trying out the demo, doesn't it? I think a better approach would be to deploy the snapshot and keep the repo reference remote? What are your thoughts on this?

File: pom.xml [hunk 2]
```diff
@@ -63,7 +63,11 @@
             <artifactId>validation-api</artifactId>
             <version>1.1.0.Final</version>
         </dependency>
-
+        <dependency>
+            <groupId>org.hibernate</groupId>
+            <artifactId>hibernate-validator</artifactId>
+            <version>5.4.2.Final</version>
+        </dependency>
```
Comment (samuelgrigolato) [inline pom.xml]: I'm not sure if this is the right approach. Do you know that happens if I add `service-validator` with this dependency in a project that already has another JSR-303 implementation (e.g. a full-fledged application server)? If not let me know so I can run some tests myself and post the results. Maybe to fail at the absence of a implementation (properly helping the developer in a friendly and intuitive way via console output) is better than pushing an implementation behind the scenes?

File: demos/spring-5-demo/build.gradle (hunk 1 above)
Comment (andrelugomes) [reply demos/spring-5-demo/build.gradle]: I agree, but this a simple example. I don't see any problem.

Comment (samuelgrigolato) [reply demos/spring-5-demo/build.gradle]: Ok, fair enough.

**Pull request merged**